libnmstate/nm/ovs.py

```diff
@@ -125,36 +124,38 @@ def create_patch_setting(patch_state):
     return patch_setting
 
 
-def is_ovs_bridge_type_id(type_id):
-    return type_id == NM.DeviceType.OVS_BRIDGE
-
-
 def is_ovs_port_type_id(type_id):
     return type_id == NM.DeviceType.OVS_PORT
 
 
-def is_ovs_interface_type_id(type_id):
-    return type_id == NM.DeviceType.OVS_INTERFACE
-
-
-def get_port_by_port(nmdev):
-    active_con = connection.get_device_active_connection(nmdev)
-    if active_con:
-        controller = active_con.get_controller()
-        if controller and is_ovs_port_type_id(controller.get_device_type()):
-            return controller
-    return None
+def get_ovs_bridge_info(nm_dev_ovs_br):
+    iface_info = {OB.CONFIG_SUBTREE: {}}
+    ovs_info = iface_info[OB.CONFIG_SUBTREE]
+    ports_info = _get_bridge_nmstate_ports_info(nm_dev_ovs_br)
+    options = _get_bridge_options(nm_dev_ovs_br)
 
+    if ports_info or options:
+        ovs_info[OB.PORT_SUBTREE] = ports_info
+        ovs_info[OB.OPTIONS_SUBTREE] = options
+    return iface_info
 
-def get_ovs_info(context, bridge_device, devices_info):
-    port_profiles = _get_port_profiles(bridge_device, devices_info)
-    ports = _get_bridge_ports_info(context, port_profiles, devices_info)
-    options = _get_bridge_options(context, bridge_device)
 
-    if ports or options:
-        return {"port": ports, "options": options}
-    else:
-        return {}
+def _get_bridge_nmstate_ports_info(nm_dev_ovs_br):
+    ports_info = []
+    for nm_dev_ovs_port in nm_dev_ovs_br.get_slaves():
+        port_info = {}
+        nm_dev_ovs_ifaces = nm_dev_ovs_port.get_slaves()
+        if not nm_dev_ovs_ifaces:
+            continue
+        if len(nm_dev_ovs_ifaces) == 1:
+            port_info[OB.Port.NAME] = nm_dev_ovs_ifaces[0].get_iface()
+        else:
+            port_info = _get_lag_nmstate_port_info(nm_dev_ovs_port)
```

Comment:
How this works? I think this functions is returning `None`.

Reply:
There is a `return ports_info` below.